cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/replication/PullReplicatorTest.java

```diff
@@ -313,4 +313,16 @@ public void replicatorBuilderUnknownProtocol() throws URISyntaxException {
         p.build();
     }
 
+    @Test
+    public void replicatorBuilderAddsIamInterceptor() throws Exception {
+        String apiKey = "abc123";
+        ReplicatorBuilder.Pull p = ReplicatorBuilder.pull().from(new URI("http://example.com/path")).
+                to(documentStore).
+                iamApiKey(apiKey);
+        // although the replicator isn't used, the interceptor check expects the presence of the
+        // header interceptor, which only gets added if build() is called
+        ReplicatorImpl r = (ReplicatorImpl) p.build();
+        assertIamCookieInterceptorPresent(p);
+    }
+
```

Comment:
We should also have a test that specifies a URI with credentials and IAM to ensure the behaviour is as we expect i.e. only the IAM interceptor is added. The assertions in `assertIamCookieInterceptorPresent` checking the number of interceptors and the type is sufficient so the merely needs to be a copy of this with a user info https URI.

Reply:
Done in 4f18822d